File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/SpacerComponent.java
package io.github.notenoughupdates.moulconfig.gui.component;

import io.github.notenoughupdates.moulconfig.gui.GuiComponent;
import io.github.notenoughupdates.moulconfig.gui.GuiImmediateContext;

import java.util.function.Supplier;

public class SpacerComponent extends GuiComponent {
    private final Supplier<Integer> width;
    private final Supplier<Integer> height;

    public SpacerComponent(Supplier<Integer> width, Supplier<Integer> height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public int getWidth() {
        return width.get();
    }

    @Override
    public int getHeight() {
        return height.get();
    }

    @Override
    public void render(GuiImmediateContext context) {
    }
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/SpacerComponent.kt
package io.github.notenoughupdates.moulconfig.gui.component

import io.github.notenoughupdates.moulconfig.gui.GuiComponent
import io.github.notenoughupdates.moulconfig.gui.GuiImmediateContext
import java.util.function.Supplier

class SpacerComponent(
    val width: Supplier<Int>,
    val height: Supplier<Int>,
) : GuiComponent() {
    override fun getWidth(): Int {
        return width.get()
    }

    override fun getHeight(): Int {
        return height.get()
    }

    override fun render(context: GuiImmediateContext) {
    }
}